src/components/form/form.ts

```diff
@@ -0,0 +1,164 @@
+import { css, html, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { Constructor } from '../common/mixins/constructor';
+
+export interface IgcFormEventMap {
+  igcSubmit: CustomEvent<FormData>;
+  igcReset: CustomEvent;
+}
+
+/**
+ * The form is a component used to collect user input from
+ * interactive controls.
+ *
+ * @element igc-form
+ *
+ * @slot - Default slot for the form.
+ *
+ * @fires igcSubmit - Emitted when the form is submitted.
+ * @fires igcReset - Emitted when the form is reset.
+ */
+export class IgcFormComponent extends EventEmitterMixin<
+  IgcFormEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  private _controlsWithChecked = [
+    'input',
+    'radio',
+    'igc-radio',
+    'igc-switch',
+    'igc-checkbox',
+  ];
+  private _controlsWithValue = ['input', 'igc-input', 'textarea'];
+  private _controlsThatSubmit = ['input', 'button', 'igc-button'];
+
+  /** Specifies if the form should not to be validated on submit. */
```

Comment:
Let's change this to something like "Specifies if form data validation should be skipped on submit."